### PR TITLE
Rename and refactor test_spatial_kernels.py

### PR DESCRIPTION
**Pull request by hakonsbm**
This PR renames test_spatial_kernels.py to test_spatial_distributions.py as we are using distributions instead of kernels. It also refactors calculation-heavy parts of the test to make it more efficient.

### CODE REVIEW

File: pynest/nest/tests/test_spatial/test_spatial_distributions.py [hunk 1]
```diff
@@ -247,234 +286,101 @@ def _build(self):
...
+        """Calculate the probability density function in 3D"""
+        if D <= self._L_half:
+            return max(0., min(1., self._distribution(D))) * self.four_pi * D ** 2.
+        elif self._L_half < D <= self._L / math.sqrt(2):
+            return max(0., min(1., self._distribution(D))) * self.two_pi * D * (3. * self._L - 4. * D)
+        elif self._L / math.sqrt(2) < D <= self._max_dist:
+            A = self.four_pi * D ** 2.
+            C = self.two_pi * D * (D - self._L_half)
+            alpha = math.asin(1. / math.sqrt(2. - self._L ** 2. /
+                                             (2. * D ** 2.)))
```
Comment (stinebuu) [inline pynest/nest/tests/test_spatial/test_spatial_distributions.py:372]: Is it possible to have all on one line?

File: pynest/nest/tests/test_spatial/test_spatial_distributions.py [hunk 2]
```diff
@@ -162,30 +152,79 @@ def __init__(self, seed, dim, L, N, kernel_name, kernel_params=None,
...
         self._conndict = {'rule': 'pairwise_bernoulli',
-                          'p': kernel,
+                          'p': distribution,
                           'mask': maskdict}
 
+    def _calculate_constants(self, spatial_distribution):
+        """Calculate constant variables used when calculating distributions and pdfs"""
+        # Constants for spatial distribution functions
+        if spatial_distribution == 'gaussian':
+            self.two_sigma2 = 2. * self._params['sigma']**2
```
[the remark this answers is not in the document]
Comment (hakonsbm) [reply pynest/nest/tests/test_spatial/test_spatial_distributions.py:186]: When calculating the PDFs, the distribution functions (`_gauss`, `_gamma`, etc.) are called a lot of times. Variables that don't rely on the distance, `D`, are calculated in this function to make calculation of the PDFs more efficient. I have added a comment about it.

File: pynest/nest/tests/test_spatial/test_spatial_distributions.py [hunk 3]
```diff
@@ -162,30 +152,79 @@ def __init__(self, seed, dim, L, N, kernel_name, kernel_params=None,
...
+        elif spatial_distribution == 'gamma':
+            self.kappa_m_1 = self._params['kappa'] - 1
+            self.gamma_kappa_mul_theta_pow_kappa = (scipy.special.gamma(self._params['kappa']) *
```
[the remark this answers is not in the document]
Comment (hakonsbm) [reply pynest/nest/tests/test_spatial/test_spatial_distributions.py:189]: Same as above.

File: pynest/nest/tests/test_spatial/test_spatial_distributions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (stinebuu) [reply pynest/nest/tests/test_spatial/test_spatial_distributions.py:186]: Ok!

**Comment by stinebuu** [comment]
@jhnnsnk Do you have time to review this or should I assign another? :blush:

**Review by gtrensch — APPROVED** [review]
Thanks for this nice contribution 👍. Approved.

**Pull request merged**